src/provider.ts

```diff
@@ -139,9 +145,115 @@ function provider(this: any, options: ProviderOptions) {
   }
 
 
+
+  function makeUtils(utilopts: ProviderUtilityOptions, config?: any) {
+    function makeUrl(suffix: string, q: any) {
```

Comment:
place the function definitions outside the main provider function
this way we ensure there no accidental variable closures, which could introduce hidden state

Reply:
So the provider's function is being exported by default in the `provider.ts` file, maybe it's a silly question, but how would I export the `makeUtils` function if it's outside the `provider` ? And if I were to call `makeUtils` within the provider, how would I pass the utilopts and config parameters?

```
    export default provider
    if ('undefined' !== typeof (module)) {
      module.exports = provider
    }
```